src/Kunstmaan/UtilitiesBundle/Helper/Shell/Shell.php

```diff
@@ -2,39 +2,54 @@
 
 namespace Kunstmaan\UtilitiesBundle\Helper\Shell;
 
+use Symfony\Component\Process\Process;
+
 /**
  * A wrapper class which makes it possible to execute shell commands in the background.
  */
 class Shell implements ShellInterface
 {
 
     /**
-     * @param string $command  The command
-     * @param int    $priority The priority
+     * @param string $command The command
+     * @param int|null $priority The priority
      *
      * @return string The process id
+     * @throws \Symfony\Component\Process\Exception\LogicException
+     * @throws \Symfony\Component\Process\Exception\RuntimeException
      */
     public function runInBackground($command, $priority = 0)
     {
-        if ($priority) {
-            $pid = shell_exec("nohup nice -n $priority $command > /dev/null & echo $!");
-        } else {
-            $pid = shell_exec("nohup $command > /dev/null & echo $!");
-        }
+        $command = null !== $priority ?
+            sprintf('%d %s', (int) $priority, $command) :
```

Comment:
Does the priority still do anything? I think this syntax will crash because you are just sending a number to the terminal now and not to nohup. We can probably ignore the priority for now and remove it from function definition in 5.0 branch

Reply:
@Numkil, you have right, fixed. Also added BC Breaks to description.